### DISCUSSION TITLE
Keine Anzeige neuer Geräte-Firmware

### DISCUSSION THREAD

**Discussion by calllib** [OpenCCU/OpenCCU#1627] [🇩🇪 Fragen&Antworten] [answered]
Die RaspberryMatic zeigt mir keine verfügbaren Geräte-Firmware-Updates an:
![RaspberryMatic-Startseite](https://user-images.githubusercontent.com/16517367/147686372-b6b1c8e6-c9c5-429d-811b-f720dfa5c71f.png)
Die Geräte-Firmware-Seite zeigt unterschiedliche Stände für gleiche Geräte, was vielleicht nicht der Normalzustand sein sollte, und wenn ich die Versionen mit denen vergleiche, die ich in Screenshots anderer Nutzer sehe, scheinen mir manche sehr veraltet:
![GerätelisteGekürzt](https://user-images.githubusercontent.com/16517367/147686481-d27a7834-0264-41de-9ced-83e7843bd31d.png)

Ich habe dem Server vor über einem Tag den Internetzugang freigeschaltet, ihn durchgestartet und sehe auch Verbindungen auf der Firewall.

Frage 1: Warum zeigt mir die RaspberryMatic keine verfügbaren Geräte-Firmware-Updates an?
Frage 2: Wie oft oder wann sollte sie danach auf welchen Servern suchen? (Wenn ich das wüsste, könnte ich im Netz gezielter nach den Verbindungen suchen.)
Frage 3: Workaround: Finde ich die Geräte-Firmware auf einem offiziellen Website zum manuellen Download?

**Comment by Baxxy13** [comment, marked as answer]
Zu 1: weil es auf dem abgefragten eQ-3 Firmware Server keine (oder keine neuere) Firmware für deine Geräte gibt
Zu 2: man braucht eigentlich gar nicht mehr suchen, die Chancen das für die alten HM-Geräte noch neue Firmwares erscheinen sind ziemlich gering
Zu 3: [siehe hier](https://homematic-forum.de/forum/viewtopic.php?f=65&t=55168&p=549160#p549160)

**Reply by calllib** [reply]
Zu 1: Der Befehl `curl "http://ccu3-update.homematic.com/firmware/api/firmware/search/DEVICE?product=HM-CCU3"` zeigt mir (leider nur) für HM-CC-RT-DN eine neuere Firmware (1.5.3) als die installierte (1.5). Warum wird mir die auf der RaspberryMatic-Startseite nicht angeboten?
Zu 2: Die Antwort von Baxxy13 stimmt mich traurig, aber dennoch wüsste ich gerne (wegen 1): Wie oft oder wann sollte die RaspberryMatic nach Updates suchen?
Zu 3: Vielen Dank dafür!

**Reply by Baxxy13** [reply]
> Warum wird mir die auf der RaspberryMatic-Startseite nicht angeboten?

Vermutlich hast du schon die 1.5.3 drauf, die Zentrale zeigt aber nur zweistellig (1.5) an. Eine 1.5 gab es auch nie (nur 1.5.1 / 1.5.2 / 1.5.3)

> Wie oft oder wann sollte die RaspberryMatic nach Updates suchen?

Du bräuchtest dir darüber keinen Kopf machen wenn die Zentrale dauerhaft Internet-Zugang hätte. Dann schaut sie regelmäßig selbst (wenn du dich auf die WebUI einloggst und auf der Startseite bist) nach Updates.

**Reply by jens-maus** [reply]
> Zu 1: Der Befehl `curl "http://ccu3-update.homematic.com/firmware/api/firmware/search/DEVICE?product=HM-CCU3"` zeigt mir (leider nur) für HM-CC-RT-DN eine neuere Firmware (1.5.3) als die installierte (1.5). Warum wird mir die auf der RaspberryMatic-Startseite nicht angeboten?

Das ist leider ein Problem der komischen/kaputten Versionsnomenklatur der Gerätefirmwares. Die angezeigten 1.5 Firmware des HM-CC-RT-DN ist die momentan aktuellste. Wenn man die 1.5.3 runterlädt und unter der CCU/RaspberryMatic installiert sieht man da im ChangeLog das dort die `1.5.003` Versionsnummer vermerkt ist und die führende nullen hier wohl komisch interpretiert wird. Der Punkt ist jedoch das es keine neuere als die 1.5 bzw. 1.5.3 gibt und wenn deine HM-CC-RT-DN diese melden bzw. das auch in der WebUI angezeigt wird ist das so i.O.

**Reply by calllib** [reply]
Wie kann ich die installierte Version herausbekommen, wenn sie mir nur 2-stellig angezeigt wird?

**Reply by jens-maus** [reply]
> Wie kann ich die installierte Version herausbekommen, wenn sie mir nur 2-stellig angezeigt wird?

Gar nicht. Einfach so akzeptieren wie es ist. Wenn die WebUI keine neue Gerätefirmware bekannt gibt, gibt es auch keine neuere.

**Reply by calllib** [reply]
Dass mir Geräte-Firmware-Versionen nur 2-stellig angezeigt werden, ist dann halt zu akzeptieren, aber sehr betrüblich – zumal es die RaspberryMatic ja intern wohl mit mehr Stellen hinbekommt.

Wenn mir keine neue Gerätefirmware angezeigt wird, kann es aber auch daran liegen, dass der Zugriff der RaspberryMatic auf http://ccu3-update.homematic.com/firmware/api/ an irgendetwas scheitert. Also ist die fehlende Anzeige von Updates da keine Aussage über deren Nichexistenz. Vielleicht könnte auf der Startseite in diesem Fall angezeigt werden: "Update-Checks gescheitert." oder ähnlich. Nicht als Service-Meldung, denn es kann ja durchaus Absicht sein, der RaspberryMatic keinen Internetzugang zu geben, sondern eher einfach als Status.
